persistence-modules/hibernate-enterprise/src/test/java/com/baeldung/hibernate/exception/HibernateExceptionUnitTest.java

```diff
@@ -444,4 +441,21 @@ public void givenExistingEntity_whenIdUpdated_thenHibernateException() {
             closeSessionQuietly(session);
         }
     }
+
+    @Test
+    public void givenEnumParam_whenExecutingQuery_thenThrowSemanticException() {
+        thrown.expectCause(isA(SemanticException.class));
+        thrown.expectMessage("Cannot compare left expression of type");
+
+        Session session = sessionFactory.openSession();
+        session.createQuery("FROM User u WHERE u.role = 'ADMIN'", User.class);
+    }
+
+    @Test
+    public void givenEnumParam_whenSettingEnumParam_thenSemanticExceptionIsNotThrown() {
+        Session session = sessionFactory.openSession();
+        session.createQuery("FROM User u WHERE u.role = ?1", User.class)
+            .setParameter(1, Role.ADMIN);
```

Comment:
To better convey that this code does not throw an exception we can explict state it:
```suggestion
        Session session = sessionFactory.openSession();
        Assertions.assertDoesNotThrow(() -> session.createQuery("FROM User u WHERE u.role = ?1", User.class)
            .setParameter(1, Role.ADMIN));
        session.close();
```

Reply:
Done